projects/gateway2/helm/gloo-gateway/templates/gateway/proxy-deployment.yaml

```diff
@@ -1,5 +1,24 @@
 {{- $gateway := .Values.gateway }}
 {{- $statsConfig := $gateway.stats }}
+{{- $glooMtls := dict -}}
+{{- if $gateway.glooMtls }}
+  {{-  $glooMtls = $gateway.glooMtls -}}
+{{- end }} {{/* if $gateway.glooMtls.enabled */}}
+
+{{- if or $glooMtls.enabled $glooMtls.renderSecret }}
+apiVersion: v1
+kind: Secret
```

Comment:
why do we need to render this here, won't the secret already exist (as created by the mtls certgen cronjob)?

Reply:
ah i see your response above 
> This is because you can't mount secrets across namespaces.